iOS/Flows/Conversation/Conversation List/ConversationListViewController.swift

```diff
@@ -303,14 +301,31 @@ class ConversationListViewController: ViewController, ConversationListCollection
                 self?.messageInputAccessoryView.textView.setPlaceholder(for: users, isReply: false)
                 self?.messageInputAccessoryView.updateSwipeHint(shouldPlay: true)
             }
-        } else {
-            ConversationsManager.shared.activeConversation = nil
 
-            if self.isFirstResponder {
-                self.resignFirstResponder()
+            self.firstResponderTask = Task { [weak self] in
+                await Task.sleep(seconds: 0.25)
+
+                guard !Task.isCancelled else { return }
+
+                // The input accessory view should be shown when centered on a conversation. If there's not
+                // already set as first responder, then make the VC first responder.
+                if UIResponder.firstResponder.isNil {
+                    self?.becomeFirstResponder()
+                }
             }
+        } else {
+            ConversationsManager.shared.activeConversation = nil
 
             self.messageInputAccessoryView.updateSwipeHint(shouldPlay: true)
+
+            self.firstResponderTask = Task {
+                await Task.sleep(seconds: 0.25)
+
+                guard !Task.isCancelled else { return }
+
+                // Hide the keyboard and accessory view when we're not centered on a conversation.
+                UIResponder.firstResponder?.resignFirstResponder()
```

Comment:
The textfield might be the first responder so I have to resign whatever is current.